packages/ember-application/lib/system/engine.js

```diff
@@ -499,6 +499,9 @@ function commonSetupRegistry(registry) {
 
   registry.injection('view:-outlet', 'namespace', 'application:main');
 
+  registry.injection('view', '_environment', '-environment:main');
+  registry.injection('route', '_environment', '-environment:main');
```

Comment:
These injections should be done in the `engine-instance` (because the environment can change per `.visit` call).